strawberry/annotation.py

```diff
@@ -82,8 +81,6 @@ def resolve(self) -> Union[StrawberryType, type]:
             return self.create_list(evaled_type)
         elif self._is_optional(evaled_type):
             return self.create_optional(evaled_type)
-        elif self._is_scalar(evaled_type):
-            return evaled_type
```

Comment:
Can you explain why the annotation resolver no longer needs to check if an object is a scalar type? How will this work when `StrawberryScalar` is implemented?

Reply:
This check wasn't really doing much anyway because all that happens now is that scalar types fall to the bottom of this function and just get returned. The `is_scalar` check now relies on the schema context to resolve correctly so I think any kind of conversion should be pushed to schema conversion stage IMO. 

More broadly I can see us relying more and more on schema context when converting the raw python annotations into Strawberry types (and then into GraphQL-core types) so I think conversion should happen as late as possible. I came across this problem when doing #1100 and I think it will also affect issues that we are seeing with deferred annotations.